lstchain/scripts/lstchain_create_run_summary.py

```diff
@@ -0,0 +1,216 @@
+        n_pedestal = sum(
+            1 for event in source if event.trigger.event_type == EventType.SKY_PEDESTAL
+        )
+        n_sky = sum(
+            1 for event in source if event.trigger.event_type == EventType.SUBARRAY
+        )
+
+    # FIXME: Do this classification in some other way?
+    if n_sky / n_events > 0.999:
+        run_type = "DRS4"
+    elif n_pedestal / n_events > 0.1:
+        run_type = "CALI"
+    elif n_pedestal / n_events < 0.1:
+        run_type = "DATA"
+    else:
+        run_type = "UNKW"
+
+    return run_type
+
+
+def get_initial_timestamps(run_number):
+    """
+    Get initial valid timestamps. Write down the Dragon module used.
+    First subrun needs to be open. Taken from ctapipe_io_lst.event_time
+    """
+    filename = args.R0_PATH / args.date / f"LST-1.1.Run{run_number:05d}.0000.fits.fz"
+
+    with EventSource(input_url=filename) as source:
+
+        for event in source:
+            lst = event.lst.tel[tel_id]
+            central_module_index = np.where(lst.svc.module_ids == CENTRAL_MODULE)[0][0]
+            module_index = central_module_index
+            ucts_available = lst.evt.extdevices_presence & 2
+
+            # Look for the first ucts timestamp available
+            if ucts_available:
+                dragon_reference_source = "ucts"
+                dragon_reference_timestamp = lst.evt.ucts_timestamp
+                initial_dragon_counter = (
+                    int(1e9) * lst.evt.pps_counter[module_index]
+                    + 100 * lst.evt.tenMHz_counter[module_index]
+                )
+
+                return (
+                    dragon_reference_source,
+                    dragon_reference_timestamp,
+                    initial_dragon_counter,
+                    module_index,
+                )
+
+            else:
+                # No UCTS available in the first subrun
+                dragon_reference_source = "run_date"
+                dragon_reference_timestamp = lst.svc.date
+                initial_dragon_counter = (
+                    int(1e9) * lst.evt.pps_counter[module_index]
+                    + 100 * lst.evt.tenMHz_counter[module_index]
+                )
+
+                return (
+                    dragon_reference_source,
+                    dragon_reference_timestamp,
+                    initial_dragon_counter,
+                    module_index,
+                )
+
+
+def write_run_summary_to_file(date, run_numbers, list_of_run_objects):
+    """
+    Write to a .txt file the following information per run:
+    Run number
+    Number of subruns starting from 0
+    Type of run
+    Start_of_the_run
+    Reference_source ("ucts" or "run_date")
+    Reference_timestamp
+    Initial dragon counter
+    Dragon module ID used to take the counter values
+    """
+    # TODO: Be able to create file incrementally run-by-run
+    with open(args.output_dir / f"RunSummary_{date}.txt", "a+") as summary_file:
```

Comment:
Can we use a more structured file format? CSV with heade information by just using `astropy.table.Table.write`

Reply:
Sure

Reply:
I think it is done, but have a look, please